app/src/main/java/ch/epfl/sdp/peakar/user/services/providers/firebase/FirebaseAccountDataFactory.java

```diff
@@ -0,0 +1,225 @@
+package ch.epfl.sdp.peakar.user.services.providers.firebase;
+
+import android.net.Uri;
+import android.util.Log;
+
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseReference;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import ch.epfl.sdp.peakar.database.Database;
+import ch.epfl.sdp.peakar.general.remote.RemoteOutcome;
+import ch.epfl.sdp.peakar.general.remote.RemoteResource;
+import ch.epfl.sdp.peakar.points.CountryHighPoint;
+import ch.epfl.sdp.peakar.points.POIPoint;
+import ch.epfl.sdp.peakar.user.challenge.Challenge;
+import ch.epfl.sdp.peakar.user.services.AccountData;
+import ch.epfl.sdp.peakar.user.services.AuthAccount;
+import ch.epfl.sdp.peakar.user.services.AuthService;
+
+/**
+ * This class implements helper methods that can generate an AccountData retrieving data from the DB.
+ */
+public class FirebaseAccountDataFactory implements RemoteResource {
+    private final AccountData accountData;
+
+    private final DatabaseReference dbRefUser;
+    private DataSnapshot data;
+
+    private FirebaseAccountDataFactory(AccountData accountData, DatabaseReference dbRefUser) {
+        this.accountData = accountData;
+        this.dbRefUser = dbRefUser;
+    }
+
+    public static RemoteOutcome retrieveAccountData(AccountData accountData, DatabaseReference dbRefUser) {
+        FirebaseAccountDataFactory newFactory = new FirebaseAccountDataFactory(accountData, dbRefUser);
+        return newFactory.retrieveData();
+    }
+
+    @Override
+    public RemoteOutcome retrieveData() {
+        Task<DataSnapshot> retrieveTask = dbRefUser.get();
+
+        try {
+            // Wait for task to finish
+            Tasks.await(retrieveTask);
+            // Get the obtained data
+            DataSnapshot data = retrieveTask.getResult();
+            assert data != null;
+            // If there is no data on DB, return such outcome
+            if(!data.exists()) {
+                return RemoteOutcome.NOT_FOUND;
+            }
+            // Otherwise, update the attributes with retrieve data
+            this.data = data;
+            loadData();
+            return RemoteOutcome.FOUND;
+
+        } catch (Exception e) {
+            Log.d("FirebaseAccountDataFactory", "retrieveData: fail");
+            return RemoteOutcome.FAIL;
+        }
+    }
+
+    public void loadData() {
```

Comment:
function name is kind of self explainatory, but javadoc is missing

Reply:
Added doc.